app/models/post.rb

```diff
@@ -1,2 +1,17 @@
 class Post < ApplicationRecord
+  belongs_to :user, foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  after_save :increment_user_posts_count
+
+  def last_5_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def increment_user_posts_count
+    user.increment!(:posts_counter)
+  end
```

Comment:
- [ ] We will need to change the `user` here to `author` also. 

```suggestion
  def increment_user_posts_count
    author.increment!(:posts_counter)
  end
```